client/src/unifycr.c

```diff
@@ -2266,9 +2307,42 @@ int unifycr_unmount(void)
         unifycr_fd_stack = NULL;
     }
 
+    /* free resources allocated in client_rpc_init */
+    unifycr_client_rpc_finalize(&unifycr_rpc_context);
+
+    /* clean up configuration */
+    int tmp_rc = unifycr_config_fini(&client_cfg);
+    if (tmp_rc) {
+        rc = UNIFYCR_FAILURE;
+    }
+
+    /* no longer initialized, so update the flag */
+    unifycr_initialized = 0;
+
+    return rc;
+}
+
+/**
+ * unmount the mounted file system
+ * TODO: Add support for unmounting more than
+ * one filesystem.
+ * @return success/error code
+ */
+int unifycr_unmount(void)
+{
     /* invoke unmount rpc */
     printf("calling unmount\n");
     int ret = unifycr_client_unmount_rpc_invoke(&unifycr_rpc_context);
+
+    unifycr_finalize();
+
+    /* free memory tracking our mount prefix string */
+    if (unifycr_mount_prefix != NULL) {
+        free(unifycr_mount_prefix);
+        unifycr_mount_prefix = NULL;
+        unifycr_mount_prefixlen = 0;
+    }
+
```

Comment:
@adammoody I wonder if we should put this in the unmount rpc function?

Reply:
I placed things to match the clean up code so that things allocated in the left function were freed in the corresponding right function:

init --> finalize
mount --> unmount
rpc_init --> rpc_finalize

It makes sense to me for us to free this prefix string in unmount.

If we do that, we should probably also move the strdup from init to mount.

Reply:
@adammoody it might be nice to have all of the corresponding mount functionality in the mount/unmount rpc functions? Will that make it easier to organize?